tensorboard/loader.py

```diff
@@ -411,6 +413,265 @@ def __str__(self):
     return u'BufferedRecordReader{%s}' % self.path
 
 
+class RateCounter(object):
+  """Utility class for tracking how much a number increases each second.
+
+  The rate is calculated by averaging of samples within a time window,
+  which weights recent samples more strongly.
+  """
+
+  def __init__(self, window, clock=time.time):
+    """Creates new instance.
+
+    Args:
+      window: The maximum number of seconds across which rate is
+          averaged. In practice, the rate might be averaged over a time
+          period greater than window if set_value is being called less
+          frequently than window.
+      clock: Function returning a float with the number of seconds
+          since the UNIX epoch in zulu time.
+
+    :type window: float
+    :type clock: () -> float
+    """
+    self._window = window
+    self._clock = clock
+    self._points = collections.deque()
+    self._last_value = None  # type: float
+    self._last_time = None  # type: float
+
+  def get_rate(self):
+    """Determines rate of increase in value per second averaged over window.
+
+    Returns:
+      An integer representing the rate or None if not enough
+      information has been collected yet.
+
+    :rtype: int
+    """
+    points = []
+    total_elapsed = 0.0
+    total_weight = 0.0
+    for rate, elapsed, _ in self._points:
+      weight = 1.0 / (total_elapsed + 1) * elapsed
```

Comment:
Also fine with me! This now has the interesting property that it is unit-sensitive. (None of these observations is necessarily a problem: this is a heuristic, after all. I can support any reasonable heuristic that you choose as long as you qualitatively like the output.)